tchannel-core/src/main/java/com/uber/tchannel/handlers/InitRequestHandler.java

```diff
@@ -46,14 +53,19 @@ public void messageReceived(ChannelHandlerContext ctx, Message message) throws E
                 InitRequest initRequestMessage = (InitRequest) message;
 
                 if (initRequestMessage.getVersion() == InitMessage.DEFAULT_VERSION) {
-                    ChannelFuture f = ctx.writeAndFlush(new InitResponse(
+                    InitResponse initResponse = new InitResponse(
                             initRequestMessage.getId(),
-                            InitMessage.DEFAULT_VERSION,
-                            initRequestMessage.getHeaders()
-                    ));
+                            InitMessage.DEFAULT_VERSION
+                    );
+                    initResponse.setHostPort(this.channelManager.getHostPort());
+                    // TODO: figure out what to put here
+                    initResponse.setProcessName("java-process");
```

Comment:
I think this should be the name of the process that sends the request. E.g., JsonClient ...